Update the upgrade guide for Query classes

markstory opened the pull request:
Add the upgrade notes for the new query subclasses to help with forwards compatibilty. I'll update all the code examples later.

markstory committed:
Update the upgrade guide for Query classes

Add the upgrade notes for the new query subclasses to help with forwards
compatibilty. I'll update all the code examples later.

othercorey commented on en/appendices/4-5-migration-guide.rst:
```diff
@@ -32,6 +32,35 @@ features::
+
+Each of these classes no longer offer methods that don't make sense for that
+query type. For example, ``DeleteQuery`` has no ``select()`` clause. In 4.5,
+calling ``select()`` on a query intended to be used as a delete.
```
This is missing part of the sentence.

markstory replied:
There are a few nonsensical things here. I will try to do better in the future.

othercorey commented on en/appendices/4-5-migration-guide.rst:
```diff
@@ -32,6 +32,35 @@ features::
+query type. For example, ``DeleteQuery`` has no ``select()`` clause. In 4.5,
+calling ``select()`` on a query intended to be used as a delete.
+
+To help you adopt the new query interfaces and define the kind of query you want
```
```suggestion
To help you adopt the new query API and define the kind of query you want
```

othercorey commented on en/appendices/4-5-migration-guide.rst:
```diff
@@ -32,6 +32,35 @@ features::
+
+To help you adopt the new query interfaces and define the kind of query you want
+earlier. To encourage planning ahead, ``Table::query()`` is deprecated.
+Replacing it are new methods on ``Table``. The ``deleteQuery()``,
```
```suggestion
The new ``deleteQuery()``,
```

markstory committed:
Add missing ref.

markstory committed:
Fix wording.

markstory committed:
Update usage docs as well.

I realize now that I need to backport the Connection and Database
Query APIs as well.

markstory commented:
Going to hold off on merging this until I get the `Connection` methods added.
